pkgs/fake_async/lib/fake_async.dart

```diff
@@ -207,23 +210,25 @@ class FakeAsync {
   /// The [timeout] controls how much fake time may elapse before a [StateError]
   /// is thrown. This ensures that a periodic timer doesn't cause this method to
   /// deadlock. It defaults to one hour.
-  void flushTimers(
-      {Duration timeout = const Duration(hours: 1),
-      bool flushPeriodicTimers = true}) {
+  void flushTimers({
+    Duration timeout = const Duration(hours: 1),
+    bool flushPeriodicTimers = true,
+  }) {
     final absoluteTimeout = _elapsed + timeout;
     _fireTimersWhile((timer) {
       if (timer._nextCall > absoluteTimeout) {
         // TODO(nweiz): Make this a [TimeoutException].
         throw StateError('Exceeded timeout $timeout while flushing timers');
       }
 
-      if (flushPeriodicTimers) return _timers.isNotEmpty;
```

Comment:
The list is never empty here, it always contains at least `timer`, which is never removed until it is `_fire`d.